### PR TITLE
Broaden search for Fault Tolerance fallback methods

### PR DESCRIPTION
**Pull request by Azquelt**
* allow the fallback method to be private or protected
* allow the fallback method to be on a superclass or interface
* add tests for validation
* add tests for calling fallback methods in valid locations

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Failures in the build do not appear to be related.

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                    continue;
+                }
+            } else if (!Modifier.isPublic(modifiers) && !Modifier.isProtected(modifiers)) {
+                if (!getClass(classToSearch).getPackage().equals(originalClass.getPackage())) {
+                    // Ignore package-scoped methods outside of the original class's package
+                    continue;
+                }
+            }
+
+            Type[] typesToMatch = originalMethod.getGenericParameterTypes();
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java]: How do you handle varargs? Add a test

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            Type[] types = m.getGenericParameterTypes();
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java]: How do you handle parameter types with wildcard generics?

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+     * @return the class
+     * @throws RuntimeException if Type does not represent something that can be converted to a {@link Class}
+     */
+    private static Class<?> getClass(Type type) {
+        if (type instanceof Class) {
+            return (Class<?>) type;
+        } else if (type instanceof ParameterizedType) {
+            return getClass(((ParameterizedType) type).getRawType());
+        }
+        throw new RuntimeException("Could not convert " + type + " to a class, it is a " + type.getClass());
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/MethodFinder.java]: NLS and use a FT Runtime Exception

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
@OpenLiberty/message-reviewer can this be reviewed by someone please?

**Comment by Azquelt** [comment]
The personal build is a bit messy, but none of the failures appear to be related to the changes.

**Pull request merged**